nixos/modules/installer/tools/nixos-generate-config.pl

```diff
@@ -84,6 +84,15 @@ sub debug {
 }
 
 
+# nixpkgs.system
+my ($status, @systemLines) = runCommand("nix-instantiate --impure --eval --expr builtins.currentSystem");
+if ($status != 0 || join("", @systemLines) =~ /error/) {
+    die "Failed to retrieve current system type from nix.\n";
+}
+my $system = substr(@systemLines[0], 0, -1);
+push @attrs, "nixpkgs.system = lib.mkDefault $system;";
```

Comment:
I think it's more correct to set `localSystem`?

Reply:
Hm, you have a point, but `localSystem` isn't correct either, when `crossSystem` is set.
So the thing is, we can't really specify the host system with a single option.

What we want to set is the "host" system.

|Variable |Meaning when straight  | Meaning when cross|
--- | --- | ---|
|system|host|build|
|localSystem|host|build|
|crossSystem|null|host|

So there's no variable for `host`.

Damn, for a minute I thought this change was going to be simple.

Reply:
I'd be fine with `nixos-generate-config` assuming you're _not_ cross-compiling. It can also add a comment to the gen'd code saying "If you're cross-compiling, make sure to set crossSystem", or something like that.

Reply:
Ok, I've added new options.

|Variable |Meaning when straight  | Meaning when cross|
--- | --- | ---|
|hostSystem|host|host|
|buildSystem|build (=host)|build|
|LEGACY system|host|build|
|LEGACY localSystem|host|build|
|LEGACY crossSystem|null|host|

Reply:
Didn't get your message before my previous comment somehow.

> I'd be fine with `nixos-generate-config` assuming you're _not_ cross-compiling. It can also add a comment to the gen'd code saying "If you're cross-compiling, make sure to set crossSystem", or something like that.

The existing options are too complicated and less meaningful than the general / Nixpkgs terminology. We ought to fix that.

I appreciate that you try to reduce scope creep, but in this case doing half the work results in the generation of half-assed config state that resides on user machines. `hardware-configuration.nix` should either be a legacy one without any `nixpkgs.*` options, or one with `nixpkgs.hostPlatform`. Supporting a third variation would be painful.

Reply:
Do you mean to change this line after everything else?

Reply:
> I appreciate that you try to reduce scope creep, but in this case doing half the work results in the generation of half-assed config state that resides on user machines. hardware-configuration.nix should either be a legacy one without any nixpkgs.* options, or one with nixpkgs.hostPlatform. Supporting a third variation would be painful.

I was mostly trying to unblock you, without asking you to do a bunch of extra work :)

> The existing options are too complicated and less meaningful than the general / Nixpkgs terminology. We ought to fix that.

I agree, excited about the changes you made!